### DISCUSSION TITLE
`DETRHead.cls_out_channels` and `FocalLoss.use_sigmoid`

### DISCUSSION THREAD

**Discussion by ShunchiZhang** [open-mmlab/mmdetection#8821] [Q&A]
Hi, would you mind illustrate why `self.cls_out_channels = num_classes + 1` when not using sigmoid?

https://github.com/open-mmlab/mmdetection/blob/9d3e162459590eee4cfc891218dfbb5878378842/mmdet/models/dense_heads/detr_head.py#L134-L137

Besides, why `use_sigmoid = False` is not supported in focal loss class? How could I use focal loss with softmax?

https://github.com/open-mmlab/mmdetection/blob/9d3e162459590eee4cfc891218dfbb5878378842/mmdet/models/losses/focal_loss.py#L188

Thank you so much!

**Comment by zachcoleman** [comment]
In general, when `use_sigmoid=True` you are doing the binary cross-entropy per each class individually. Since every probability from each channel is [0,1] you do **not** need a background class. When `use_sigmoid=False` you are doing softmax typically and the probabilities across each class sum to 1. Thus, you **need** a background class for **not** any class.

As far as your other question. It appears it just hasn't been implemented yet.
